### DISCUSSION TITLE
BifroMq Multi-tenant global tenant, that can subscribe and publish to all topics

### DISCUSSION THREAD

**Discussion by saraheem** [apache/bifromq#111] [Q&A]
In multi tenant scenario, is there a concept of a global tenant user? that can publish and subscribe to all tenants?

**Comment by popduke** [comment]
No. The concept of ‘tenant’ in BifroMQ is simply a **namespace** for MQTT connections. It is flat rather than hierarchical. The idea of a ‘global tenant user’ is more of a management requirement, and it is designed to be addressed through HTTP APIs (e.g., HTTP Pub/Sub/Retain) and Plugin Integrations (e.g., Event Collector).

**Reply by saraheem** [reply]
so in a scenario, where I would want to receive messages from all tenants and publish them to Kafka, how would that be achieved? Events?

**Reply by popduke** [reply]
You can develop a custom event collector plugin, write all `Disted` events to kafka.

**Comment by saraheem** [comment]
Yea tried that works like a charm.  It would be really cool if there was a streaming grpc service for all the operations the http api provides + grpc streaming subscribe.

Thanks

On Dec 18, 2024, at 6:30 AM, Yonny(Yu) Hao ***@***.***> wrote:

﻿

You can develop a custom event collector plugin, write all Disted events to kafka.
